docs/architecture/blazor-for-web-forms-developers/project-structure.md

```diff
@@ -1,9 +1,233 @@
+
+```cshtml
+@page "/"
+@namespace BlazorApp1.Pages
+@addTagHelper *, Microsoft.AspNetCore.Mvc.TagHelpers
+
+<!DOCTYPE html>
+<html lang="en">
+<head>
+    <meta charset="utf-8" />
+    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
+    <title>BlazorApp1</title>
+    <base href="~/" />
+    <link rel="stylesheet" href="css/bootstrap/bootstrap.min.css" />
+    <link href="css/site.css" rel="stylesheet" />
+</head>
+<body>
+    <app>
+        @(await Html.RenderComponentAsync<App>(RenderMode.ServerPrerendered))
+    </app>
+
+    <script src="_framework/blazor.server.js"></script>
+</body>
+</html>
+```
+
+In the Blazor WebAssembly app, the host page is a simple static HTML file under *wwwroot/index.html*. The `<app>` element is used to indicate where the root component should be rendered.
+
+```html
+<!DOCTYPE html>
+<html>
+<head>
+    <meta charset="utf-8" />
+    <meta name="viewport" content="width=device-width" />
+    <title>BlazorApp2</title>
+    <base href="/" />
+    <link href="css/bootstrap/bootstrap.min.css" rel="stylesheet" />
+    <link href="css/site.css" rel="stylesheet" />
+</head>
+<body>
+    <app>Loading...</app>
+
+    <script src="_framework/blazor.webassembly.js"></script>
+</body>
+</html>
+```
+
+The specific component to render is configured in the app's `Startup.Configure` method with a corresponding CSS selector for where the component should be rendered.
+
+```csharp
+    public class Startup
+    {
+        public void ConfigureServices(IServiceCollection services)
+        {
+        }
+
+        public void Configure(IComponentsApplicationBuilder app)
+        {
+            app.AddComponent<App>("app");
+        }
+    }
+```
+
+## Build output
+
+When the Blazor projects are built, all of the Razor component files are compiled along with the project code into a single assembly. Unlike ASP.NET Web Forms projects, Blazor doesn't support runtime compilation of the UI logic.
+
+## Running the app
+
+To run the Blazor Server app simply hit F5 in Visual Studio. Because Blazor apps do not support runtime compilation, to see the results of code changes, including component markup changes, you typically need to rebuild and restart the app when running under the debugger. If you don't run under the debugger (Ctrl+F5), Visual Studio watches for file changes and restarts the app as changes are made. You simply refresh the browser as changes are made.
```

Comment:
```suggestion
To run the Blazor Server app, press `F5` in Visual Studio. Because Blazor apps do not support runtime compilation, to see the results of code changes, including component markup changes, you typically need to rebuild and restart the app when running under the debugger. If you don't run under the debugger (`Ctrl+F5`), Visual Studio watches for file changes and restarts the app as changes are made. You simply refresh the browser as changes are made.
```